scaffolding-chef-inspec/lib/scaffolding.sh

```diff
@@ -205,7 +205,8 @@ EOF
         "token": "{{cfg.automate.token}}",
         "node_name": "{{ sys.hostname }}",
         "verify_ssl": false{{#if cfg.automate.environment}},
-        "environment": "{{cfg.automate.environment}}"{{/if }}
+        "environment": "{{cfg.automate.environment}}"{{/if }}{{#if cfg.automate.node_uuid}},
+        "node_uuid": "{{cfg.automate.node_uuid}}"{{/if }}
```

Comment:
@mattray Thanks for putting this in. I have a question how does this work with the `target_id` section set? One of the things that I have been striving for with the interaction with Automate is that both chef and inspec use the habitat supervisor machine_id so that when all three are reporting in they show up as one node. I think this probably needs to be an if or around the `target_id` in the JSON and should also be the same for windows.

Reply:
It appears that the `node_uuid` overrides the `target_id`, which was the previous source of the uuid inside of Automate. I don't see target_id (or node_uuid) documented in https://www.inspec.io/docs/reference/config/ or in https://github.com/inspec/inspec/blob/master/docs/config.md, but if that's the canonical ID for the InSpec node I'd be willing to modify the patch. Let's see if we can get someone from the @chef/inspec-maintainers to weigh-in.

Reply:
InSpec itself has no ID for nodes - it is passed only the connection details for _some target_ as input, and it doesn't maintain any state about what that target is. 
To date, InSpec results have been pinned to an Automate node worked via the `audit` cookbook, which supplements InSpec's report data with a `node_uuid` [it gets from the Chef run](https://github.com/chef-cookbooks/audit/blob/master/libraries/helper.rb#L20) :

```
# Returns the node's uuid
  def entity_uuid
    # the Chef::DataCollector::Messages API here is Chef < 15.0 backcompat and can be removed when Chef 14.x is no longer supported
    node['chef_guid'] || defined?(Chef::DataCollector::Messages) && Chef::DataCollector::Messages.node_uuid
  end
```

I've not seen `target_id` as an alternative to `node_uuid` before.

Reply:
After this conversation, I believe this patch is the right approach for adding nodes to Automate.